File: Matury CKE/2022 Czerwiec/Programowanie/C++/zad4_2.cpp
#include <iostream>
#include <string>
#include <fstream>
#include <vector>
#include <algorithm>
#include <cmath>

int reflection(std::string str)
{
    std::reverse(str.begin(), str.end());
    return std::stoi(str);
}

int main()
{
    std::cout << "4.2. ";

    // std::ifstream dane("C:/Matury-Informatyka/Matury CKE/2022 Czerwiec/DANE/przyklad.txt");
    std::ifstream dane("C:/Matury-Informatyka/Matury CKE/2022 Czerwiec/DANE/liczby.txt");

    std::vector<std::string> numsList;
    std::vector<int> reflectedAndDivBy17List;

    int refDifMax, numMax = 0;

    std::string num;
    while(dane >> num){
        numsList.push_back(num);
        int numNow = std::stoi(num);
        int refNum = reflection(num);

        int difNow = abs(numNow-refNum);
        if(difNow > refDifMax){
            refDifMax = difNow;
            numMax = numNow;
        }
    }

    std::cout << numMax << " " << refDifMax;

    return 0;
}

File: Matury CKE/2022 Czerwiec/Programowanie/Python/zad4_2.py
import math
# numbers=[i.strip() for i in open('../../DANE/przyklad.txt').readlines()]
numbers=[i.strip() for i in open('../../DANE/liczby.txt').readlines()]

max_diff=0
max_num=0
for i in numbers:
    reflection=int(i[::-1])
    num_diff=int(i)-reflection
    if math.fabs(num_diff)>max_diff:
        max_diff=int(math.fabs(num_diff))
        max_num=i

print(f'4.2. {max_num} {max_diff}')